src/nnet3/nnet-am-decodable-simple.cc

```diff
@@ -203,15 +199,79 @@ void NnetDecodableBase::GetCurrentIvector(int32 output_t_start,
                 << " (mismatched --ivector-period?)";
     }
     ivector_frame = online_ivector_feats_->NumRows() - 1;
+  } else if (ivector_frame < 0) {
+    ivector_frame = 0;
   }
-  *ivector = online_ivector_feats_->Row(ivector_frame);
+  ivector->CopyFromVec(online_ivector_feats_->Row(ivector_frame));
 }
 
+void NnetDecodableBase::GetIvectorsForFrames(int32 output_t_start,
+                                             int32 num_output_frames,
+                                             Matrix<BaseFloat> *ivectors) {
+  // if no ivectors have been specified either as online ivectors or as ivector
+  // for utterance, just return and leave the size of ivectors matrix as 0 
+  if (ivector_ == NULL && online_ivector_feats_ == NULL) {
+    KALDI_ASSERT(ivectors->NumCols() == 0);
+    return;
+  }
+  const int32 left_context = nnet_left_context_ + opts_.extra_left_context,
```

Comment:
Here the extra_left_context is taken into account to give ivectors to the extra-left-context